extensions/guacamole-auth-jdbc/modules/guacamole-auth-jdbc-base/src/main/java/org/apache/guacamole/auth/jdbc/tunnel/RestrictedGuacamoleTunnelService.java

```diff
@@ -193,6 +200,12 @@ public int compare(ModeledConnection a, ModeledConnection b) {
         // Return the first unreserved connection
         for (ModeledConnection connection : sortedConnections) {
 
+            // If connection weight is zero or negative, this host is disabled and should not be used.
+            if (connection.getConnectionWeight() < 1) {
+                logger.warn("Weight for {} is < 1, connection will be skipped.", connection.getName());
```

Comment:
Perhaps this should instead be logged at the debug level?

If zero and negative weights are intended mechanisms for disabling connection, logging this at the warning level will result in the logs being filled with messages warning the admin that they're using a feature as intended.